eventbrite.py

```diff
@@ -0,0 +1,430 @@
+            "error_detail": error_detail,
+            "status_code": resp.status
+        }
+    data = resp.data.decode('utf-8')
+    return json.loads(data)
+
+def delete_event(http, event_id):
+    url = f"{EVENTBRITE_API_BASE_URL}events/{event_id}/"
+    headers = {
+    'Authorization': eventbrite_api_token,
+    'Content-Type': 'application/json'
+    }
+
+    # Make the DELETE request
+    resp = http.request('DELETE', url, headers=headers)
+
+    if resp.status != 200:
+        error_data = json.loads(resp.data.decode('utf-8'))
+        error_detail = error_data.get('error_detail', {})
+        return {
+            "error": error_data.get('error', 'Unknown Error'),
+            "error_description": error_data.get('error_description', 'An error occurred'),
+            "error_detail": error_detail,
+            "status_code": resp.status
+        }
+    
+    # A successful DELETE request should return a 200 status code for Eventbrite
+    return {"message": "Event deleted successfully", "status_code": resp.status}
+
+def unpublish_event(http, event_id):
+    url = f"{EVENTBRITE_API_BASE_URL}events/{event_id}/unpublish/"
+    headers = {
+    'Authorization': eventbrite_api_token,
+    'Content-Type': 'application/json'
+    }
+    # Make the POST request
+    resp = http.request('POST', url, headers=headers)
+
+    # Check the status code to ensure a successful request
+
+    if resp.status != 200:
+        error_data = json.loads(resp.data.decode('utf-8'))
+        error_detail = error_data.get('error_detail', {})
+        return {
+            "error": error_data.get('error', 'Unknown Error'),
+            "error_description": error_data.get('error_description', 'An error occurred'),
+            "error_detail": error_detail,
+            "status_code": resp.status
+            }
+    data = resp.data.decode('utf-8')
+    return {"message": "Event unpublished successfully", "status_code": resp.status}, json.loads(data)
+    
+def publish_event(http, event_id):
+    url = f"{EVENTBRITE_API_BASE_URL}events/{event_id}/publish/"
+    headers = {
+    'Authorization': eventbrite_api_token,
+    'Content-Type': 'application/json'
+    }
+
+    # Make the POST request
+    resp = http.request('POST', url, headers=headers)
+
+    if resp.status != 200:
+        error_data = json.loads(resp.data.decode('utf-8'))
+        error_detail = error_data.get('error_detail', {})
+        return {
+            "error": error_data.get('error', 'Unknown Error'),
+            "error_description": error_data.get('error_description', 'An error occurred'),
+            "error_detail": error_detail,
+            "status_code": resp.status
+            }
+    data = resp.data.decode('utf-8')
+    return json.loads(data)
+
+
+def create_ticket_class(http, event_id, ticket_details):
+    url = f"{EVENTBRITE_API_BASE_URL}events/{event_id}/ticket_classes/"
+    headers = {
+    'Authorization': eventbrite_api_token,
```

Comment:
Don't treat `eventbrite_api_token` as a global, pass it through the function:
```py
def create_ticket_class(http, eventbrite_api_token, event_id, ticket_details)
```
Apply to rest of the functions.